docs/sphinx/contributing/contributing.rst

```diff
@@ -0,0 +1,150 @@
+
+To develop code for Marvin you will first need a `GitHub account <https://github.com/>`__. Then go to the `Marvin GitHub page <https://github.com/sdss/marvin/>`__ and `fork it <https://help.github.com/articles/fork-a-repo/>`__. Develop your changes in a new branch and, when ready, open a `pull request <https://help.github.com/articles/about-pull-requests/>`__. Pleade, make sure the pull request describes all the changes the code introduces. Before it can be merged into master, yoir pull request needs to be approved by at least one of the repository owners.
+
+Testing
+^^^^^^^
+
+All new code must be properly tested and contain enough unittests. No new code will be accepted that does not meet this requirement. Tests must run in, at least, Python 2.7 and 3.6.
+
+All new pull requests will trigger a `Travis <https://travis-ci.org/>`__ build. Unfortunaly, Marvin's CI system is broken at the time and most Travis runs will fail (most likely due to timeouts). We hope to fix this soon but in the meantime it is still useful to check the logs of the build, since your tests probably did run.
+
+For local testing, you will need to set up a PostgreSQL server with a test database called ``manga``, restored from this `dump file <https://sas.sdss.org/marvin/data/travis_mangadb.sql>`__. Then run the command ``run_marvin --debug``, which will create a local flask HTTP server. You can now go to the ``python/marvin/tests`` directory and run ``pytest``, which will run all tests (fair warning, it may take a while!) or ``pytests <your-file>``.
+
+.. _marvin-contributing-code-documentation:
+
+Documentation
+^^^^^^^^^^^^^
+
+If your changes introduce a new feature or change the behaviour of user-facing commands, you will need to provide Sphinx documentation on how to use it. Marvin's documentation can be found in the `docs/ directory <https://github.com/sdss/marvin/tree/master/docs>`__. Add your documentation where you think it is appropriate. As part of the pull request review, you may be asked to restructure or move the documentation.
+
+
+Contributing documentation
+--------------------------
+
+Code is only useful if it is well documented! An excellent way of contributing to Marvin is by writing documentation. There are different forms of documentation that you can submit:
+
+* Sphinx documentation of current features, as described in :ref:`marvin-controbuting-code-documentation`.
+* Plain-text documentation. If you do not want to write Sphinx documentation, you can still send us the text and we will format it appropriately. Send your text in the form of a `new issue`_.
+* Open a `new issue`_ pointing out parts of the documentation that are unclear, misleading, or deprecated.
+* Write :ref:`marvin-tutorials <tutorials>`! Tutorials can be in Sphinx or ipyhton notebook format.
+* If you have used Marvin in the classroom and developed activities based on it we would love to hear about it!
+
+
+Contributing a VAC
```

Comment:
We should make sure this fits in with the real `Contributing a VAC` guidelines the Data Team have and make it clear this is optional for people with MaNGA VACS.  It should also include a timeline placed on the VAC owners, similar to the DR timeline, that places these steps in context relative to those.    

This documentation implies a lot of steps needed for VAC inclusion.  I think that's fine and necessary, but it may be unreasonable now to expect VAC owners to deliver this content on a timeline for inclusion in DR15.  Especially given writing Python-code, tests, Sphinx documentation, will be all new to most of these people.  And given the Data Team's usual problems with corralling VAC owners to provide what the "simple things" they need for the DR release.  This will require extra time for us to vet everything, which I doubt we have the time and/or manpower for.

Reply:
I honestly don't think we need to require tests or documentation for the VACs. Maybe that's not clear in the documentation. Honestly, as long as we make clear the VACs are not officially supported by us or MaNGA, and they don't break anything else, I'd leave it to the VAC owners. Chances of us getting more than a couple VACs otherwise are very remote.

Also, not sure we need to tie this to any DR timeline. We are not even including VACs in the database and there is no server counterpart for this. A new VAC can be implemented in any new release of Marvin.

Reply:
VACs are officially supported.  Not officially maintained but MaNGA VACs have to get official approval and vetting by the DPC before they are allowed to be released.  And they have to adhere to the official VAC guidelines by the Data Team before they are allowed to be released.  Documentation is one of the mandatory guidelines.  We should be the same.  

Maybe we don't need to require tests but that's the only real way to ensure they don't break anything else.  If we can write some simple example tests on the `helloworld` example they can emulate, that might be enough.  

VACs are only released during Data Releases and not in-between.  Even internally.  I think we should adopt a similar policy.  New VACs will only be included during a release of Marvin for the Data Release when the VACs themselves are being released.  It keeps things a bit cleaner because that's when VAC owners are preparing their VACs.  Even if we roll out VACs in releases after a DR, I think it's better to impose a relative timeline.  Otherwise, people will never do it or take forever.  I think Anne-Marie would agree with me here that people need deadlines placed on them.